attribution/token_perturbation.py

```diff
@@ -1,48 +1,68 @@
-from typing import List
+from typing import List, Optional
 
 import numpy as np
 from sklearn.neighbors import NearestNeighbors
-from transformers import PreTrainedTokenizer
+from transformers import GPT2LMHeadModel, GPT2Tokenizer, PreTrainedTokenizer
 
 FIXED_REPLACEMENT_TOKEN = " "
 
 
-def get_replacement_token(
-    token_id_to_replace: int,
-    perturbation_strategy: str,
-    embeddings: np.ndarray = None,
-    tokenizer: PreTrainedTokenizer = None,
-) -> int:
-    if perturbation_strategy == "fixed":
-        return tokenizer.encode(FIXED_REPLACEMENT_TOKEN, add_special_tokens=False)[0]
-    elif perturbation_strategy == "distant":
-        return get_increasingly_distant_token_ids(
-            token_id_to_replace, embeddings, n_tokens=4
-        )[-1]
-    elif perturbation_strategy == "nearest":
-        return get_most_similar_token_ids(
-            token_id_to_replace, embeddings, tokenizer, 2
-        )[1]
-    else:
-        raise ValueError(f"Unknown perturbation strategy: {perturbation_strategy}")
+class PerturbationStrategy:
+    def get_replacement_token(self, token_id_to_replace: int) -> int:
+        raise NotImplementedError
+
+
+class FixedPerturbationStrategy(PerturbationStrategy):
```

Comment:
I think this class is mixing `self.replacement_token` with the global `FIXED_REPLACEMENT_TOKEN`.

Should probably make sure it's using `self.replacement_token` and use `FIXED_REPLACEMENT_TOKEN` as the argument default value